File: SourceCppProject/sources/Parsers/StringsCommentsHandlerTest.cpp
#include <gtest/gtest.h>

#include "StringsCommentsHandler.hpp"


TEST(StringsCommentsHandlerTest, CutAllCommentsInsideString_TestCase_1)
{
    std::string artifact {"0 1 /* 1 */ 2 /* 2 */ 3 4 5 6 7 /* 7 */ 8 /* 8 */ 9"};

    const std::string actual_artifact {StringsCommentsHandler(artifact).trimWrappedCommentsInsideString()};
    const std::string expected_artifact {"0 1  2  3 4 5 6 7  8  9"};

    EXPECT_EQ(actual_artifact, expected_artifact);
}

TEST(StringsCommentsHandlerTest, CutAllCommentsInsideString_TestCase_2)
{
    std::string artifact {"0 1 /*step1*/ 2 /*step2*/ 3 /*step3*/ 4 5"};

    const std::string actual_artifact {StringsCommentsHandler(artifact).trimWrappedCommentsInsideString()};
    const std::string expected_artifact {"0 1  2  3  4 5"};

    EXPECT_EQ(actual_artifact, expected_artifact);
}

TEST(StringsCommentsHandlerTest, CutAllCommentsInsideString_TestCase_3)
{
    std::string artifact {"/* comment */   0 1 2 3 4 5"};

    const std::string actual_artifact {StringsCommentsHandler(artifact).trimWrappedCommentsInsideString()};
    const std::string expected_artifact {"   0 1 2 3 4 5"};

    EXPECT_EQ(actual_artifact, expected_artifact);
}

TEST(StringsCommentsHandlerTest, CutAllCommentsInsideString_TestCase_4)
{
    std::string artifact {"0 1 2 3 4 5   /* comment */\n"};

    const std::string actual_artifact {StringsCommentsHandler(artifact).trimWrappedCommentsInsideString()};
    const std::string expected_artifact {"0 1 2 3 4 5   \n"};

    EXPECT_EQ(actual_artifact, expected_artifact);
}

TEST(StringsCommentsHandlerTest, CutAllCommentsInsideString_TestCase_5)
{
    std::string artifact {"/* comment */0 1 2 /*   comment   */ 3 4 5/* comment */\n"};

    const std::string actual_artifact {StringsCommentsHandler(artifact).trimWrappedCommentsInsideString()};
    const std::string expected_artifact {"0 1 2  3 4 5\n"};

    EXPECT_EQ(actual_artifact, expected_artifact);
}

File: TextLinesComparing.Testing/StringsCommentsHandlerTest.cs
using TextLinesComparing.Library;

namespace TextLinesComparing.Testing;

public class StringsCommentsHandlerTest
{
    [Test]
    public void TrimSingleStringComment_TestCase()
    {
        Assert.Multiple(() =>
        {
            Assert.That(
            actual: StringsCommentsHandler.TrimSingleStringComment(
                artifactString: "This is a test artifact # with a comment"),
            expression: Is.EqualTo("This is a test artifact "));

            Assert.That(
                actual: StringsCommentsHandler.TrimSingleStringComment(
                    artifactString: "012345#6789"),
                expression: Is.EqualTo("012345"));

            Assert.That(
                actual: StringsCommentsHandler.TrimSingleStringComment(
                    artifactString: "# This is a comment string!"),
                expression: Is.EqualTo(""));
        });
    }

    [Test]
    [Ignore("SKIP this test until fixing errors in TrimSingleStringComment() implementation")]
    public void CutAllCommentsInsideString_TestCase_1()
    {
        const string artifact = "0 1 /* 1 */ 2 /* 2 */ 3 4 5 6 7 /* 7 */ 8 /* 8 */ 9";

        string actual_artifact = StringsCommentsHandler.TrimWrappedCommentsInsideString(artifact);
        string expected_artifact = "0 1  2  3 4 5 6 7  8  9";

        Assert.That(actual_artifact, Is.EqualTo(expected_artifact));
    }

    [Test]
    [Ignore("SKIP this test until fixing errors in TrimSingleStringComment() implementation")]
    public void CutAllCommentsInsideString_TestCase_2()
    {
        const string artifact = "0 1 /*step1*/ 2 /*step2*/ 3 /*step3*/ 4 5";

        string actual_artifact = StringsCommentsHandler.TrimWrappedCommentsInsideString(artifact);
        string expected_artifact = "0 1  2  3  4 5";

        Assert.That(actual_artifact, Is.EqualTo(expected_artifact));
    }

    [Test]
    [Ignore("SKIP this test until fixing errors in TrimSingleStringComment() implementation")]
    public void CutAllCommentsInsideString_TestCase_3()
    {
        const string artifact = "/* comment */   0 1 2 3 4 5";

        string actual_artifact = StringsCommentsHandler.TrimWrappedCommentsInsideString(artifact);
        string expected_artifact = "   0 1 2 3 4 5";

        Assert.That(actual_artifact, Is.EqualTo(expected_artifact));
    }

    [Test]
    [Ignore("SKIP this test until fixing errors in TrimSingleStringComment() implementation")]
    public void CutAllCommentsInsideString_TestCase_4()
    {
        const string artifact = "0 1 2 3 4 5   /* comment */\n";

        string actual_artifact = StringsCommentsHandler.TrimWrappedCommentsInsideString(artifact);
        string expected_artifact = "0 1 2 3 4 5   \n";

        Assert.That(actual_artifact, Is.EqualTo(expected_artifact));
    }

    [Test]
    [Ignore("SKIP this test until fixing errors in TrimSingleStringComment() implementation")]
    public void CutAllCommentsInsideString_TestCase_5()
    {
        const string artifact = "/* comment */0 1 2 /*   comment   */ 3 4 5/* comment */\n";

        string actual_artifact = StringsCommentsHandler.TrimWrappedCommentsInsideString(artifact);
        string expected_artifact = "0 1 2  3 4 5\n";

        Assert.That(actual_artifact, Is.EqualTo(expected_artifact));
    }
}
